org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/prolog/PrologModel.java

```diff
@@ -0,0 +1,187 @@
+			ICompletionResponse response, SharedSettings settings){
+		CompletionItem item = new AttributeCompletionItem(attrName, canSupportSnippet, editRange, generateValue,
+				defaultValue, enumerationValues, settings);
+		MarkupContent markup = new MarkupContent();
+		markup.setKind(MarkupKind.MARKDOWN);
+		markup.setValue(documentation);
+		item.setDocumentation(markup);
+		response.addCompletionItem(item);
+	}
+
+	public static void computeAttributeNameCompletionResponses(ICompletionRequest request, 
+	ICompletionResponse response, Range editRange, DOMDocument document, SharedSettings settings)
+			throws BadLocationException {
+
+		if (document.hasProlog() == false) {
+			return;
+		}
+		int offset = document.offsetAt(editRange.getStart());
+		DOMNode prolog = document.findNodeAt(offset);
+		if(!prolog.isProlog()) {
+			return;
+		}
+		boolean isSnippetsSupported = request.getCompletionSettings().isCompletionSnippetsSupported();
+		
+		if(!prolog.hasAttribute(VERSION_NAME)) {
+			createCompletionItem(VERSION_NAME, isSnippetsSupported, true, editRange, VERSION_1, VERSION_VALUES, null, response, settings);
+		}
+
+		if(!prolog.hasAttribute(ENCODING_NAME)) {
+			createCompletionItem(ENCODING_NAME, isSnippetsSupported, true, editRange, UTF_8, ENCODING_VALUES, null, response, settings);
+		}
+
+	}
+
+	public static void computeValueCompletionResponses(ICompletionRequest request, 
+			ICompletionResponse response, Range editRange, DOMDocument document, SharedSettings settings) throws BadLocationException {
+		
+		if (document.hasProlog() == false) {
+			return;
+		}
+		int offset = document.offsetAt(editRange.getStart());
+		DOMNode prolog = document.findNodeAt(offset);
+		if(!prolog.isProlog()) {
+			return;
+		}
+
+		DOMAttr attr = prolog.findAttrAt(offset);
+		if(VERSION_NAME.equals(attr.getName())) { // version
+			createCompletionItemsForValues(VERSION_VALUES, editRange, document, response, settings);
+		}
+
+		else if(ENCODING_NAME.equals(attr.getName())) { // encoding
+			createCompletionItemsForValues(ENCODING_VALUES, editRange, document, response, settings);
+		}
+	}
+
+	private static void createCompletionItemsForValues(Collection<String> enumerationValues, Range editRange, DOMDocument document, ICompletionResponse response, SharedSettings settings) {
+		
+		// Figure out which quotation to use for filter text
+		String settingQuotation = settings.formattingSettings.getQuotationAsString();
+		String currentQuotation = settingQuotation;
+		try {
+			int start = document.offsetAt(editRange.getStart());
+			int end = document.offsetAt(editRange.getEnd());
+			if(start != end) {
+				currentQuotation = String.valueOf(document.getText().charAt(start));
+			}
+		} catch (BadLocationException e) {
+		}
+
+		CompletionItem item;
+		for (String option : enumerationValues) {
+			String optionWithQuotes = settingQuotation + option + settingQuotation;
+			item = new CompletionItem();
+			item.setLabel(option);
```

Comment:
you need to set a sortText value so that the CompletionItems order stays consistent with the enumeration order.
See https://github.com/eclipse/eclipse.jdt.ls/blob/fc6b40e58808d23a324628a1e79e50f117c534de/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/SortTextHelper.java and its usage